<?xml version="1.0"?>
<launch>
  <arg name="input_type"      default="video" /> <!-- video, bag, or live_camera -->
  <arg name="zed_name"        default="zed_objdetect" />
  <arg name="zed_node_name"   default="zed_objdetect" />
  <arg name="camera_model"    default="zed2" />
  <arg name="detection_algorithm" default="CONTOURS" />

  <include file="$(find zed_wrapper)/launch/include/zed_camera.launch.xml" if="$(eval input_type != 'video')">
    <arg name="common_params_file"   value="$(find controller_node)/config/zed_objdet/common.yaml" />
    <arg name="camera_params_file"   value="$(find controller_node)/config/zed_objdet/$(arg camera_model).yaml" />
    <arg name="node_name"            value="$(arg zed_node_name)" />
    <arg name="publish_urdf"         value="true" />
    <arg name="camera_name"          value="$(arg zed_name)" />
    <arg name="camera_model"         value="$(arg camera_model)" />
    <arg name="base_frame"           value="base_link" />
    <arg name="cam_pos_x"            value="0.2665" />
    <arg name="cam_pos_y"            value="0.0" />
    <arg name="cam_pos_z"            value="0.4917" />
    <arg name="cam_roll"             value="0.0" />
	  <arg name="cam_pitch"            value="0.0" />
    <arg name="cam_yaw"              value="0.0" />
  </include>

  <group ns="tf_object_detection">
  <!--
    <node pkg="tf_object_detection" name="tf_object_tracker" type="tracker.py" output="screen">
    </node>
  -->
	  <rosparam command="load" file="$(find tf_object_detection)/config/tf_default_config.yaml" />
    <node pkg="tf_object_detection" name="tf_object_detection" type="trtdetection.py" output="screen">
    </node>
    <!--
    <node pkg="tf_object_detection" name="check_depth" type="check_depth.py" output="screen">
    </node>
    -->
    <node pkg="tf_object_detection" name="tf_object_screen_to_world" type="tf_object_screen_to_world" output="screen">
      <param name="depth_algorithm" type="string" value="$(arg depth_algorithm)" />
    </node>
    <node pkg="tf_object_detection" name="cargo_detection_filter" type="cargo_detection_filter" output="screen">
    </node>
  

	   <!--Possible that this should not be in namespace?-->
    <node pkg="tf_object_detection" name="pub_video" type="pub_video.py" output="screen" if="$(eval input_type == 'video')">
      <rosparam command="load" file="$(find tf_object_detection)/config/pub_video_config.yaml" />
    </node>
  </group>
  <arg name="cam_image_topic" default="left/image_rect_gray" />
	<arg name="cam_info_topic" default="left/camera_info" />
	<arg name="output_frame" default="/zed_ar_left_camera_frame" />

	<!--
	<include file="$(find apriltag_launch)/launch/apriltag_continuous_detection.launch">
		<arg name="camera_name"            value="$(arg zed_name)" />
		<arg name="image_topic"            value="$(arg cam_image_topic)" />
	</include>
	-->
  <group ns="goal_detection">
    <node pkg="apriltag_launch" name="apriltag_to_pf_node" type="apriltag_to_pf_node" output="screen">
    </node>
  </group>
  
  <!-- Launch apriltags-->
  <include file="$(find cuda_apriltag_ros)/launch/cuda_continuous_detection.launch">
	</include>
</launch>
